### ISSUE TITLE
Fix false positives for maps and double-slash comments in indentation

### ISSUE THREAD

**Issue by tvooo** [stylelint/stylelint#2946]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where the "indentation" rule throws false positives when using a double-slash comment on the last line of a multi-level SCSS map.

> Which rule, if any, is this issue related to?

`indentation`

> What CSS is needed to reproduce this issue?

The following SCSS showcases the issue:

```scss
/* False positives */

$my-map1: (
  'map2': (
    'key1': value1, // This comment is great
    'key2': value2, // This comment causes indentation error in line 8
  ),
);

$my-map2: (
  'map2': (
    'key1': value1, // This comment is great
    'key2': value2,
    // This comment causes indentation error in line 16
  ),
);

/* All good */

$my-map3: (
  'map2': (
    'key1': value1, // This comment is great
    'key2': value2, // This comment does NOT cause indentation error, due to the blank line below?

  ),
);

$my-map4: (
  'map2': (
    'key1': value1, // This comment is great
    'key2': value2,
  ),
);

$my-map5: (
  'map2': (
    'key1': value1, // This comment is great
    'key2': value2, /* Not using double-slash comments works as well! */
  ),
);
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "indentation": 2
  }
}
```

> Which version of stylelint are you using?

e.g. `8.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config .stylelintrc.js --syntax scss $(find src/scss -path '*.scss')`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS maps and SCSS double-slash comments

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged, given the example code above:

```
8:1 error Expected indentation of 2 spaces (indentation)
16:1 error Expected indentation of 2 spaces (indentation)
```

**Comment by jeddy3** [comment]
@tvooo Thanks for using the template and for the very thorough write-up!

I'm going to label this as "help wanted". @tvooo Please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).  You'll find the `indentation` rule code [here](https://github.com/stylelint/stylelint/blob/master/lib/rules/indentation/index.js).

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.
